OpenRA.Game/Graphics/IGraphicsDevice.cs

```diff
@@ -15,23 +15,10 @@
 
 namespace OpenRA
 {
-	[AttributeUsage(AttributeTargets.Assembly)]
-	public sealed class PlatformAttribute : Attribute
-	{
-		public readonly Type Type;
-
-		public PlatformAttribute(Type graphicsDeviceType)
-		{
-			if (!typeof(IDeviceFactory).IsAssignableFrom(graphicsDeviceType))
-				throw new InvalidOperationException("Incorrect type in RendererAttribute");
-			Type = graphicsDeviceType;
-		}
-	}
-
-	public interface IDeviceFactory
+	public interface IPlatform
 	{
 		IGraphicsDevice CreateGraphics(Size size, WindowMode windowMode);
-		ISoundEngine CreateSound();
+		ISoundEngine CreateSound(string device);
```

Comment:
I guess this file and it's directory location should change at one point as it contains more than `IGraphicsDevice` now.